src/Lie_algebra/Lie_algebra_interface.jl

```diff
@@ -42,23 +42,228 @@ function LieAlgebra(G::LieGroup{𝔽,O}) where {𝔽,O<:AbstractGroupOperation}
+
+Return the vector of coordinates to the decomposition of `X` with respect to an [`AbstractBasis`](@extref `ManifoldsBase.AbstractBasis`)
+of the [`LieAlgebra`](@ref) `𝔤`.
+The operation can be performed in-place of `c`.
+
+By default this function requires that [`identity_element`](@ref)`(G, T)` is available and calls
+the corresponding [`get_coordinates`](@extref ManifoldsBase :jl:function:`ManifoldsBase.get_coordinates`) function
+of the Riemannian manifold the Lie group is build on.
+
+The inverse operation is [`get_vector`](@ref).
+
+See also [`vee`](@ref).
+"""
+
+@doc "$(_doc_get_coordinates)"
+function ManifoldsBase.get_coordinates(
+    𝔤::LieAlgebra, X, B::ManifoldsBase.AbstractBasis=DefaultLieAlgebraOrthogonalBasis()
+)
+    return ManifoldsBase._get_coordinates(𝔤, X, B)
+end
+# Mimic the levels from ManifoldsBase just without the base point p
+function ManifoldsBase._get_coordinates(
+    𝔤::LieAlgebra, X::T, B::ManifoldsBase.AbstractBasis
+) where {T}
     G = 𝔤.manifold
-    return get_coordinates(base_manifold(G), identity_element(G), X, B)
+    return get_coordinates(base_manifold(G), identity_element(G, T), X, B)
 end
-function ManifoldsBase.get_coordinates!(𝔤::LieAlgebra, c, X, B::ManifoldsBase.AbstractBasis)
+@doc "$(_doc_get_coordinates)"
+function ManifoldsBase.get_coordinates!(
+    𝔤::LieAlgebra, c, X, B::ManifoldsBase.AbstractBasis=DefaultLieAlgebraOrthogonalBasis()
+)
+    return ManifoldsBase._get_coordinates!(𝔤, c, X, B)
+end
+function ManifoldsBase._get_coordinates!(
+    𝔤::LieAlgebra, c, X::T, B::ManifoldsBase.AbstractBasis
+) where {T}
     G = 𝔤.manifold
-    get_coordinates!(base_manifold(G), c, identity_element(G), X, B)
+    return ManifoldsBase.get_coordinates!(base_manifold(G), c, identity_element(G, T), X, B)
+end
+function ManifoldsBase._get_coordinates(
+    𝔤::LieAlgebra, X, B::DefaultLieAlgebraOrthogonalBasis
+)
+    return get_coordinates_lie(𝔤, X, number_system(B))
+end
+function ManifoldsBase._get_coordinates!(
+    𝔤::LieAlgebra, c, X, B::DefaultLieAlgebraOrthogonalBasis
+)
+    get_coordinates_lie!(𝔤, c, X, number_system(B))
     return c
 end
 
-function ManifoldsBase.get_vector(𝔤::LieAlgebra, c, B::ManifoldsBase.AbstractBasis)
+# the hat/vee variant
+function get_coordinates_lie(𝔤::LieAlgebra, X, N)
+    c = allocate_result(𝔤, get_coordinates, X, DefaultLieAlgebraOrthogonalBasis(N))
+    return get_coordinates_lie!(𝔤, c, X, N)
+end
+function get_coordinates_lie!(𝔤::LieAlgebra, c, X::T, N) where {T}
+    # Provide a default fallback
     G = 𝔤.manifold
-    return get_vector(base_manifold(G), identity_element(G), c, B)
+    return get_coordinates!(
+        base_manifold(G),
+        c,
+        identity_element(G, T),
+        X,
+        ManifoldsBase.DefaultOrthogonalBasis(N),
+    )
 end
-function ManifoldsBase.get_vector!(𝔤::LieAlgebra, X, c, B::ManifoldsBase.AbstractBasis)
+
+_doc_get_vector = """
+    get_vector(G::LieGroup, c, B::AbstractBasis; kwargs...)
```

Comment:
What `kwargs` does this support?

Reply:
We could try to do it implicitly as well. as we do in `Manifolds.jl` in a few places, but the main problem compared to `Manifolds.jl` is that for manifolds we have some point `p` where we are at and we can use its type to “find” (or infer) the corresponding tangent vector type. Herer we do not have that and I do expect `c` to really always just be an array of coefficients.